pkg/virt-api/webhooks/validating-webhook/admitters/migration-create-admitter_test.go

```diff
@@ -34,91 +33,65 @@ import (
 	"kubevirt.io/client-go/api"
 
 	v1 "kubevirt.io/api/core/v1"
-	"kubevirt.io/client-go/kubecli"
+	kubevirtfake "kubevirt.io/client-go/generated/kubevirt/clientset/versioned/fake"
 
 	"kubevirt.io/kubevirt/pkg/virt-api/webhooks"
 )
 
 var _ = Describe("Validating MigrationCreate Admitter", func() {
-	// Mock VirtualMachineInstanceMigration
-	var ctrl *gomock.Controller
-	var virtClient *kubecli.MockKubevirtClient
+	var virtClient *kubevirtfake.Clientset
```

Comment:
Please consider dropping the shared variable and use a client per test.
It will enable dropping the `BeforeEach` function and allow a different approach for initializing the client:
```go
virtClient := kubevirtfake.NewSimpleClientset(<objects you want to have in the cluster>)
```

For example, vmi's could already be present in the fake client, without calling their matching "Create()" method.

https://github.com/kubevirt/kubevirt/blob/05bb90e20d0a9064ff27309d67e1828030266963/pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter_test.go#L116

Reply:
Good idea, allow me to offer a slightly different approach.

How about keeping everything in `BeforeEach` and call `virtClient.KubevirtV1().VirtualMachineInstances(vmi.Namespace).UpdateStatus(context.Background(), vmi, metav1.UpdateOptions{})` when necessary (e.g. after Status updates)?

Reply:
Discussed offline, it would in fact be cleaner without `BeforeEach`.